Metrics.py

```diff
@@ -37,16 +47,323 @@ def truePosOrNeg(self, dataSet, truePosOrNeg):
 		if len(possibleClassifications) != 2:
 			return "Cannot calculate true positive or negative rate for nonbinary classifications"
 		else:
-			bothPosOrNeg = 0
-			realPosOrNeg = 0 #actual number of people with this classification in the DataSet
+			matchesLabel = 0
+			actualLabel = 0  # actual number of people with this classification in the DataSet
 
 			for i in range(dataFrame.shape[0]):
 				groundTruth = dataFrame.at[i, dataSet.trueLabels]
 				bayesClassification = dataFrame.at[i, "Bayes Classification"]
 
 				if bayesClassification == truePosOrNeg and groundTruth == truePosOrNeg:
-					bothPosOrNeg +=1
+					matchesLabel +=1
 				if groundTruth == truePosOrNeg:
-					realPosOrNeg +=1
+					actualLabel +=1
+
+			return matchesLabel, actualLabel
+
+
+	'''
+	Takes the information from truePosOrNeg and turns it into a rate
+	'''
+	def truePosOrNegRate(self, matchesLabel, actualLabel):
+		return matchesLabel / actualLabel * 100
+
+	'''
+	Segments the overarching DataSet into smaller (non-overlapping) subsets by protected attribute. Returns a list of these
+	subsets as well as a list of the corresponding protected attribute values.
+		dataSet (DataSet) - the overarching DataSet NOTE WE'RE PROBABLY GETTING RID OF THIS PART WHEN WE MAKE IT AN INSTANCE VARIABLE
+	'''
+	def determineGroups(self, dataSet):
```

Comment:
I was under the assumption that we were only using binary sensitive attributes? Is there a reason this function accounts for more than two possible protected attribute values?

Reply:
We thought that we had fixed the protectedAttributes but we messed up. Thanks!